zipkin-server/README.md

```diff
@@ -213,223 +223,330 @@ $ java -jar zipkin.jar
+You can specify the settings for one or more indexes individually by using `SW_STORAGE_ES_SPECIFIC_INDEX_SETTINGS`.
+
+**NOTE:**
+Supported settings:
+- number_of_shards
+- number_of_replicas
+
+**NOTE:** These settings have the highest priority and will override the existing
+generic settings mentioned in [index settings doc](#index-settings).
+
+The settings are in `JSON` format. The index name here is logic entity name, which should exclude the `${ZIPKIN_NAMESPACE}` which is `zipkin` by default, e.g.
+```json
+{
+  "metrics-all":{
+    "number_of_shards":"3",
+    "number_of_replicas":"2"
+  },
+  "segment":{
+    "number_of_shards":"6",
+    "number_of_replicas":"1"
+  }
+}
+```
+
+This configuration in the YAML file is like this,
+```yaml
+storage:
+  elasticsearch:
+    # ......
+    specificIndexSettings: ${ZIPKIN_STORAGE_ES_SPECIFIC_INDEX_SETTINGS:"{\"metrics-all\":{\"number_of_shards\":\"3\",\"number_of_replicas\":\"2\"},\"segment\":{\"number_of_shards\":\"6\",\"number_of_replicas\":\"1\"}}"}
+```
+
+#### Recommended ElasticSearch server-side configurations
+You could add the following configuration to `elasticsearch.yml`, and set the value based on your environment.
+
+```yml
+# In tracing scenario, consider to set more than this at least.
+thread_pool.index.queue_size: 1000 # Only suitable for ElasticSearch 6
+thread_pool.write.queue_size: 1000 # Suitable for ElasticSearch 6 and 7
+
+# When you face a query error on the traces page, remember to check this.
+index.max_result_window: 1000000
 ```
 
-Under the scenes, these map to properties prefixed `zipkin.storage.elasticsearch.ssl.`, which affect
-the Armeria client used to connect to Elasticsearch.
+We strongly recommend that you read more about these configurations from ElasticSearch's official documentation since they directly impact the performance of ElasticSearch.
 
-The above properties allow the most common SSL setup to work out of box. If you need more
-customization, please make a comment in [this issue](https://github.com/openzipkin/zipkin/issues/2774).
+### BanyanDB storage components
 
-#### Automatic Index Creation
-Zipkin will automatically create new indices as needed. Elasticsearch by default [allows](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#index-creation) automatic creation of said indices, though your local install may have been configured to disallow it. You can verify this in the cluster settings: `action.auto_create_index: false`.
+[BanyanDB](https://github.com/apache/skywalking-banyandb) is a dedicated storage implementation developed by the SkyWalking Team and the community.
+Currently, BanyanDB is still in the PoC stage and it is not recommended to use it in a production environment.
+
+The following apply when `ZIPKIN_STORAGE` is set to `banyandb`:
+    
+    * `ZIPKIN_STORAGE_BANYANDB_HOST`: The host of BanyanDB. Defaults to `127.0.0.1`.
+    * `ZIPKIN_STORAGE_BANYANDB_PORT`: The port of BanyanDB. Defaults to `17912`.
```

Comment:
These should be merged as `targets`. Could you recheck your dependency version of OAP and sync the doc?

Reply:
The configuration and documentation have been updated.